### PR TITLE
config: mark run-ddl and max-server-connections as removed

### LINKED ISSUE #37185: run-ddl should be marked as removed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

This is a small cleanup required after https://github.com/pingcap/tidb/pull/35425

The run-ddl option is still present in tidb_config:
```sql
mysql> select @@tidb_config\G
*************************** 1. row ***************************
@@tidb_config: {
..
	"index-limit": 64,
	"instance": {
..
		"tidb_enable_ddl": true,
..
	},
..
	"run-ddl": true,
..
}
1 row in set (0.01 sec)
```

With other variables converted to instance scope, the previous usage has been marked as removed, so it patches out of @@tidb_config.

### 2. What did you expect to see? (Required)

There should be a line for run-ddl in this list:
https://github.com/pingcap/tidb/blob/9709249e523a92cf635ab330f6e2a0ef77a0843a/config/config.go#L990-L1039

### 3. What did you see instead (Required)

See above

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #37185

Problem Summary:

Already described in the issue.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by xiongjiwei — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -274,10 +271,11 @@ type Config struct {
 	// OOMUseTmpStorage unused since bootstrap v93
 	OOMUseTmpStorage bool `toml:"oom-use-tmp-storage" json:"oom-use-tmp-storage"`
 
-	// CheckMb4ValueInUTF8, EnableCollectExecutionInfo, Plugin are deprecated.
```
Comment (morgo) [inline config/config.go:277]: Can we change the comment to say this contains deprecated items? That will make it clearer why they are grouped last.

**Review by morgo — APPROVED** [review]
Small nit in comment, otherwise LGTM.

**Pull request merged**